src/librustdoc/html/render/context.rs

```diff
@@ -49,36 +48,43 @@ pub(crate) struct Context<'tcx> {
     /// The current destination folder of where HTML artifacts should be placed.
     /// This changes as the context descends into the module hierarchy.
     pub(crate) dst: PathBuf,
-    /// A flag, which when `true`, will render pages which redirect to the
-    /// real location of an item. This is used to allow external links to
-    /// publicly reused items to redirect to the right location.
-    pub(super) render_redirect_pages: bool,
     /// Tracks section IDs for `Deref` targets so they match in both the main
     /// body and the sidebar.
-    pub(super) deref_id_map: DefIdMap<String>,
+    pub(super) deref_id_map: RefCell<DefIdMap<String>>,
     /// The map used to ensure all generated 'id=' attributes are unique.
-    pub(super) id_map: IdMap,
+    pub(super) id_map: RefCell<IdMap>,
     /// Shared mutable state.
     ///
     /// Issue for improving the situation: [#82381][]
     ///
     /// [#82381]: https://github.com/rust-lang/rust/issues/82381
-    pub(crate) shared: Rc<SharedContext<'tcx>>,
+    pub(crate) shared: SharedContext<'tcx>,
+    /// Collection of all types with notable traits referenced in the current module.
+    pub(crate) types_with_notable_traits: RefCell<FxIndexSet<clean::Type>>,
+    /// Contains information that needs to be saved and reset after rendering an item which is
+    /// not a module.
+    pub(crate) info: ContextInfo,
+}
+
+#[derive(Clone, Copy)]
+pub(crate) struct ContextInfo {
```

Comment:
This should have a doc comment about why it exists and when things should go here (vs in Context).